backend/server.py

```diff
@@ -226,43 +235,49 @@ def add_status():
     status_data = {}

```

Comment:
Instead of waiting in a loop until `status_data` is populated, could we return an empty `JSON` response if no new data is available?

Would returning an empty `JSON` response be a viable approach to avoid blocking behavior, and would it ensure that no unintended updates occur if no data is present?